texture_manager.c

```diff
@@ -594,17 +519,31 @@ void texture_manager_background_texture_loader(void *dummy) {
 
         while (cur_tex) {
             texture_2d *old_cur = NULL;
-
-            if (cur_tex->pixel_data == NULL && cur_tex->url != NULL && !cur_tex->failed) {
-                TEXLOG("Passing to load_image_with_c: %s", cur_tex->url);
-
-                if (!resource_loader_load_image_with_c(cur_tex)) { //if not loading from C remove from list
+            const char *url = cur_tex->url;
+
+            if (url != NULL) {
```

Comment:
This code to check whether this is a canvas is in two places. There should be a `static inline bool` function to check whether a `const char*` says `__canvas__`.